app/src/main/java/ru/myuniquenickname/myapplication/presentation/movie_list/MovieListFragment.kt

```diff
@@ -0,0 +1,232 @@
+package ru.myuniquenickname.myapplication.presentation.movie_list
+
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Bundle
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.SearchView
+import android.widget.Toast
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.work.WorkManager
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import ru.myuniquenickname.myapplication.R
+import ru.myuniquenickname.myapplication.data.work_manager.WorkRepository
+import ru.myuniquenickname.myapplication.databinding.FragmentMoviesListBinding
+import ru.myuniquenickname.myapplication.presentation.TransactionsFragmentClicks
+
+class MovieListFragment : Fragment() {
+
+    private var listener: TransactionsFragmentClicks? = null
+    private var _binding: FragmentMoviesListBinding? = null
+    private val binding get() = _binding!!
+    private val movieListViewModel: MovieListViewModel by viewModel()
+    private lateinit var adapterMovies: MoviesAdapter
+    private val workRepository = WorkRepository()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (savedInstanceState == null) {
+            movieListViewModel.refreshPopularMovies()
+        }
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentMoviesListBinding.inflate(inflater, container, false)
+        adapterMovies = MoviesAdapter(clickListener)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        WorkManager.getInstance(requireContext()).enqueue(workRepository.constrainedRequest)
```

Comment:
Why isn't it the responsibility of the ViewModel?

Reply:
I need a context to create, isn't it better to create here?